File: LivrusVirtus/wxWindows/wx.NET/VCPP.NET/ConProfissoes.cpp
#include "stdafx.h"
#include "ConProfissoes.h"
#include "Resource.h"
#include "ConsultasSQL.h"

namespace Livrus
{
  frmConProfissoes::frmConProfissoes( wx::Window* parent )
  	: frmConsultas(parent) {

    Title = Resource::STR_CONPROFISS;
    labDescricao->Label = "Profissão: ";
    gridConsulta->SetColLabelValue(1, "Profissão");

    EVT_BUTTON((int)idt::ID_FRMCONSULTAS_BTNPESQUISAR, 
      new wx::EventListener( this, OnBtnPesquisar ));
    EVT_BUTTON((int)idt::ID_FRMCONSULTAS_BTNLIMPAR, 
      new wx::EventListener( this, OnBtnLimpar ));
    EVT_BUTTON((int)idt::ID_FRMCONSULTAS_BTNFECHAR, 
      new wx::EventListener( this, OnBtnFechar ));
  	EVT_CLOSE( new wx::EventListener( this, OnClose ) );
    EVT_GRID_SELECT_CELL(new wx::EventListener( this, OnGridSelectCell ));
    EVT_GRID_LABEL_LEFT_CLICK(new wx::EventListener( this, 
      OnGridLabelLeftClick ));
  }

  Void frmConProfissoes::OnClose( System::Object* sender, wx::Event* e ){
    Fecha(Resource::MNU_CON_PROFISSOES);
  	e->Skip();
  }

  Void frmConProfissoes::OnBtnFechar( System::Object* sender, wx::Event* e ){
    Fecha(0);
    e->Skip();
  }
  
  Void frmConProfissoes::OnBtnLimpar( System::Object* sender, wx::Event* e ){
    LimparDados();
    e->Skip();
  }
  
  Void frmConProfissoes::OnBtnPesquisar( System::Object* sender, wx::Event* e ){
    PesquisarDados(ConsultasSQL::ConsSQL->Profissao('S',"", edtDescricao->Value),
      Resource::STR_PROFNENC, 2);
    e->Skip();
  }
  
  Void frmConProfissoes::OnGridSelectCell( System::Object* sender, 
    wx::Event* e ){
    wx::GridEvent* evt = dynamic_cast<wx::GridEvent*>(e);
    setLabRegistros(evt->Row);
    e->Skip();
  }
  
  Void frmConProfissoes::OnGridLabelLeftClick( System::Object* sender, 
    wx::Event* e ){
    wx::GridEvent* evt = dynamic_cast<wx::GridEvent*>(e);
    setLabRegistros(evt->Row);
    e->Skip();
  }
}

File: LivrusVirtus/wxWindows/wx.NET/Delphi.NET/fontes/ConProfissoes.pas
unit ConProfissoes;

interface

uses
  System.Drawing, wx, Consultas;

type
  TfrmConProfissoes = class(TfrmConsultas)
  private
    procedure OnClose(sender: System.Object; e: wx.Event);
    procedure OnBtnFechar(sender: System.Object; e: wx.Event);
    procedure OnBtnLimpar(sender: System.Object; e: wx.Event);
    procedure OnBtnPesquisar(sender: System.Object; e: wx.Event);
    procedure OnGridSelectCell(sender: System.Object; e: wx.Event);
    procedure OnGridLabelLeftClick(sender: System.Object; e: wx.Event);
  public
    constructor Create(parent: wx.Window);
  end;

var
  frmConProfissoes: TfrmConProfissoes;

  [assembly: RuntimeRequiredAttribute(TypeOf(TfrmConProfissoes))]

implementation

uses Resource, ConsultasSQL;

constructor TfrmConProfissoes.Create(parent: wx.Window);
begin
  inherited Create(parent, 'Profissão:');

  Title := Res.STR_CONPROFISS;
  //labDescricao.Label := 'Profissão:';
  gridConsulta.SetColLabelValue(1, 'Profissão');

  EVT_BUTTON(ID_FRMCONSULTAS_BTNPESQUISAR, OnBtnPesquisar );
  EVT_BUTTON(ID_FRMCONSULTAS_BTNLIMPAR, OnBtnLimpar );
  EVT_BUTTON(ID_FRMCONSULTAS_BTNFECHAR, OnBtnFechar );
  EVT_CLOSE( OnClose );
  EVT_GRID_SELECT_CELL( OnGridSelectCell );
  EVT_GRID_LABEL_LEFT_CLICK( OnGridLabelLeftClick );
end;

procedure TfrmConProfissoes.OnClose(sender: System.Object; e: wx.Event);
begin
  Fecha(Res.MNU_CON_PROFISSOES);
	e.Skip();
end;

procedure TfrmConProfissoes.OnBtnFechar(sender: System.Object; e: wx.Event);
begin
  Fecha(0);
  e.Skip();
end;

procedure TfrmConProfissoes.OnBtnLimpar(sender: System.Object; e: wx.Event);
begin
  LimparDados();
  e.Skip();
end;

procedure TfrmConProfissoes.OnBtnPesquisar(sender: System.Object; e: wx.Event);
begin
  PesquisarDados(ConsSQL.Profissao('S','', edtDescricao.Value),
    Res.STR_PROFNENC, 2);
  e.Skip();
end;

procedure TfrmConProfissoes.OnGridSelectCell(sender: System.Object; e: wx.Event);
var evt: wx.GridEvent;
begin
  evt := wx.GridEvent(e);
  setLabRegistros(evt.Row);
  e.Skip();
end;

procedure TfrmConProfissoes.OnGridLabelLeftClick(sender: System.Object;
  e: wx.Event);
var evt: wx.GridEvent;
begin
  evt := wx.GridEvent(e);
  setLabRegistros(evt.Row);
  e.Skip();
end;

end.
